apps/ejabberd/src/mongoose_riak_sup.erl

```diff
@@ -0,0 +1,117 @@
+%%==============================================================================
+%% Copyright 2015 Erlang Solutions Ltd.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%% http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%==============================================================================
+-module(mongoose_riak_sup).
+
+-behaviour(supervisor).
+
+%% API
+-export([start/2]).
+-export([start_link/2]).
+-export([stop/0]).
+-export([get_riak_pools_count/0]).
+
+%% Supervisor callbacks
+-export([init/1]).
+
+-define(SERVER, ?MODULE).
+
+-include("ejabberd.hrl").
+
+%%%===================================================================
+%%% API functions
+%%%===================================================================
+
+start(Workers, PoolsSpec) ->
+    ChildSpec = {?MODULE, {?MODULE, start_link, [Workers, PoolsSpec]},
+        transient, infinity, supervisor, [?MODULE]},
+    {ok, _} = supervisor:start_child(ejabberd_sup, ChildSpec).
+%%--------------------------------------------------------------------
+%% @doc
+%% Starts the supervisor
+%%
+%% @end
+%%--------------------------------------------------------------------
+-spec(start_link(integer(), list()) ->
+    {ok, Pid :: pid()} | ignore | {error, Reason :: term()}).
+start_link(Workers, PoolSpec) ->
```

Comment:
Can we remove this multi-pool logic?  It think It would simplify the logic on our side, especially situations when one of the nodes goes down or extending/shrinking the Riak cluster. 

With one pool we give one simple guarantee  - as long as I can have one pool alive I will run. If something is wrong I will go down. With multi-pool we may start thinking how many nodes down means failure, should Mongoose try to restore the dead pools after certain amount of time or how to add a new one etc.  For example, In case of MAM I doesn't make sense to serve users if we cannot store their messages.

All in all, I'm a fan of one-pool approach here and removing all possibilities to setting more that one pool. Doing that we forcing users to go in the right direction :). If somebody needs to test, he/she can do that with one Riak node or use HAproxy on localhost.  

Regarding DNS it depends on DNS server. I think that ROUTE 53 can return only 1 IP and also is able to do health checks.

I would like to hear what other guys think about that ! If it turns out that I'm alone in this opinion, I will merge it :)